BrainPortal/app/controllers/data_providers_controller.rb

```diff
@@ -97,8 +97,8 @@ def create  #:nodoc:
       flash[:notice] = "Provider successfully created."
       respond_to do |format|
         format.html { redirect_to :action => :index, :format => :html}
-        format.xml  { render :xml   => @provider }
-        format.json { render :json  => @provider }
+        format.xml  { render :xml   => @provider.for_api }
+        format.json { render :json  => @provider.for_api }
```

Comment:
While we're invoking .for_api() on these objects, shouldn't we configure the attributes that we would liked returned back to the client, based on DP type? E.g. for a UserKey DP, return the remote host, user, and path? For a S3 DP, the path, bucket name etc? (Never return any tokens of course)